subsys/net/l2/openthread/Kconfig

```diff
@@ -155,7 +155,6 @@ endmenu # "Zephyr optimizations"
 
 config OPENTHREAD_SHELL
 	bool "Enable OpenThread shell"
-	depends on !OPENTHREAD_COPROCESSOR
```

Comment:
The commit log just states:
> `OPENTHREAD_SHELL` does no longer dependend on
> `OPENTHREAD_COPROCESSOR` being disabled.

but why does shell no longer depend on `OPENTHREAD_COPROCESSOR=n` ?

What has changed. And what is the impact for v1.6.0 if this stays as it was in v1.4.0, v1.5.0, v.1.5.1, etc ?

Can OpenThread shell be used both in NCP and RCP mode ?
https://github.com/zephyrproject-rtos/zephyr/pull/28458/commits/0c0a77a22368fddb79aefb65c34c7681e824d00c

Could you please give a thorough explanation in the commit message why this is now supported.